### PR TITLE
[IMP] mail: move code to models (AttachmentViewer) 

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2996277

### CODE REVIEW

File: addons/mail/static/src/models/attachment_viewer.js [hunk 1]
```diff
@@ -153,6 +153,14 @@ registerModel({
         onClickVideo(ev) {
             ev.stopPropagation();
         },
+        /**
+         * Called when clicking on zoom in icon.
```
Comment (oomsveta) [inline addons/mail/static/src/models/attachment_viewer.js:165]: useless

File: addons/mail/static/src/models/attachment_viewer.js [hunk 2]
```diff
@@ -169,6 +169,15 @@ registerModel({
             ev.stopPropagation();
             this.zoomOut();
         },
+        /**
+         * Called when clicking on reset zoom icon.
```
Comment (oomsveta) [inline addons/mail/static/src/models/attachment_viewer.js:181]: useless

File: addons/mail/static/src/models/attachment_viewer.js [hunk 3]
```diff
@@ -23,6 +23,21 @@ registerModel({
...
+            if (!this.exists() || !this.attachmentViewerViewable) {
+                return;
+            }
+            if (
+                this.attachmentViewerViewable.isImage &&
+                (!this.imageRef || !this.imageRef.complete)
+            ) {
+                this.update({ isImageLoading: true });
+            }
+        },
```
[the remark this answers is not in the document]
Comment (alexkuhn) [reply addons/mail/static/src/models/attachment_viewer.js:50]: `imageRef.complete` is not observed, so cannot turn this into compute. I much prefer for this PR to stay with structural changes

**Comment by Xavier-Do** [comment]
@robodoo r-
This is breaking design-theme and l10n. Please take a look
https://runbot.odoo.com/runbot/build/19766203

**Comment by alexkuhn** [comment]
Re-target to master: https://github.com/odoo/odoo/pull/102046

**Pull request closed without merging**